File: BaekJoon/22859/22859.js
function solution(input) {
  let answer = [];
  input = input.replaceAll(/<\/?main>/g, "");
  const divTags = input
    .split(/<div title="(.*?)">(.*?)<\/div>/)
    .filter((v) => v !== "");
  for (let i = 0; i < divTags.length; i++) {
    if (divTags[i][0] === "<") {
      const pTags = divTags[i].split(/<p>(.*?)<\/p>/);
      for (let j = 1; j < pTags.length; j += 2) {
        answer.push(
          pTags[j]
            .replaceAll(/<\/?[a-zA-Z ]*?>/g, "")
            .trim()
            .replaceAll(/ {2,}/g, " ")
        );
      }
    } else {
      const title = divTags[i].trim();
      answer.push(`title : ${title}`);
    }
  }
  console.log(answer.join("\n").trim());
}
// const filePath = "input.txt";
const filePath = "/dev/stdin";
const line = require("fs").readFileSync(filePath, "utf8");
let input = line.trim();
solution(input);

File: BaekJoon/22859/22859.py
import re
import sys

# sys.stdin = open("input.txt", "r")

document = input()

body = re.sub(r"<\/?main>", "", document)

tokens = list(
    filter(lambda x: x != "", re.split(r'<div title="(.*?)">(.*?)<\/div>', body))
)

answer = []


def extract(token):
    if token[0] != "<":
        return answer.append(f"title : {token.strip()}")

    paragraphs_list = list(re.split(r"<p>(.*?)<\/p>", token))
    for i in range(1, len(paragraphs_list), 2):
        paragraphs = paragraphs_list[i]
        content = re.sub(r"<\/?[a-zA-Z ]*?>", "", paragraphs).strip()
        answer.append(
            re.sub(r" {2,}", " ", content)
        )


for token in tokens:
    extract(token)

print("\n".join(answer).strip())
